File: src/main/java/com/qtech/bubbles/common/VersionType.java
package com.qtech.bubbles.common;

import com.qtech.bubbles.annotation.FieldsAreNonnullByDefault;
import com.qtech.bubbles.annotation.MethodsReturnNonnullByDefault;

import javax.annotation.ParametersAreNonnullByDefault;

@FieldsAreNonnullByDefault
@ParametersAreNonnullByDefault
@MethodsReturnNonnullByDefault
public enum VersionType implements Translatable {
    ALPHA("alpha"), BETA("beta"), RELEASE("release"), CANDIDATE("rc");

    private final String name;

    VersionType(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name;
    }

    public String toRepresentation() {
        return "VersionType{" +
                "name='" + name + '\'' +
                '}';
    }

    public String getName() {
        return name;
    }

    @Override
    public String getTranslationId() {
        return "misc.version_type." + name;
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/VersionType.kt
package com.qtech.bubbles.common

import com.qtech.bubbles.annotation.FieldsAreNonnullByDefault
import com.qtech.bubbles.annotation.MethodsReturnNonnullByDefault
import javax.annotation.ParametersAreNonnullByDefault

@FieldsAreNonnullByDefault
@ParametersAreNonnullByDefault
@MethodsReturnNonnullByDefault
enum class VersionType(val text: String) : Translatable {
    ALPHA("alpha"), BETA("beta"), RELEASE("release"), CANDIDATE("rc");

    override fun toString(): String {
        return text
    }

    fun toRepresentation(): String {
        return "VersionType{" +
                "name='" + text + '\'' +
                '}'
    }

    override val translationId: String
        get() = "misc.version_type.$text"
}